autoload/ale/handlers/rust.vim

```diff
@@ -46,9 +46,11 @@ function! ale#handlers#rust#HandleRustErrorsForFile(buffer, full_filename, lines
         endif
 
         for l:span in l:error.spans
+            " substitue used for window file paths
+            let l:file_name = substitute(l:span.file_name, '\\', '\\\\', 'g')
             if (
             \   l:span.is_primary
-            \   && (a:full_filename =~ (l:span.file_name . '$') || l:span.file_name ==# '<anon>')
+            \   && (a:full_filename =~ (l:file_name . '$') || l:span.file_name ==# '<anon>')
```

Comment:
Could you try using `ale#path#IsBufferPath` for this instead? I'm trying to use the one function for handling everything. That's step 1 towards supporting setting errors from other files in quickfix. Step two is outputting `'filename': whatever` in the Dictionary objects, and then checking the filename returned automatically.